util/chunk/column.go

```diff
@@ -246,6 +241,85 @@ const (
 	sizeMyDecimal = int(unsafe.Sizeof(types.MyDecimal{}))
 )
 
+// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
+func (c *Column) preAlloc(length, typeSize int) {
+	nData := length * typeSize
+	if len(c.data) >= nData {
+		c.data = c.data[:nData]
+	} else {
+		c.data = make([]byte, nData)
+	}
+
+	nBitmap := (length + 7) >> 3
+	if len(c.nullBitmap) >= nBitmap {
+		c.nullBitmap = c.nullBitmap[:nBitmap]
+		for i := range c.nullBitmap {
+			// resets all slots to null.
+			c.nullBitmap[i] = 0
+		}
+	} else {
+		c.nullBitmap = make([]byte, nBitmap)
+	}
+
+	if c.elemBuf != nil && len(c.elemBuf) >= typeSize {
+		c.elemBuf = c.elemBuf[:typeSize]
+	} else {
+		c.elemBuf = make([]byte, typeSize)
+	}
+
+	c.length = length
+}
+
+// SetNull sets the rowIdx to null.
+func (c *Column) SetNull(rowIdx int, isNull bool) {
+	if isNull {
+		c.nullBitmap[rowIdx>>3] &= ^(1 << uint(rowIdx&7))
+	} else {
+		c.nullBitmap[rowIdx>>3] |= 1 << uint(rowIdx&7)
+	}
+}
+
+// NullCount returns the number of nulls in this Column.
+func (c *Column) NullCount() int {
```

Comment:
seems this function is unused, can we remove it?

Reply:
It is used when encoding.
https://github.com/pingcap/tidb/pull/11485/files#diff-5486cf4e3d2501e339835af7bab0b859R61